### PR TITLE
feat: add sorting functionality

### PR DESCRIPTION
**Pull request by n0ruSh**
To support sorting functionality. 

[Issue Link](https://github.com/ant-design/ant-design/issues/26414)

### CODE REVIEW

**Comment by zombieJ** [comment]
Add `filterSort` instead `sort` & `sortOrder`. It's meaningless in top level since user always can sort their self by pass `options`.

File: src/generate.tsx [hunk 1]
```diff
@@ -187,7 +188,7 @@ export interface GenerateConfig<OptionsType extends object[]> {
   /** Convert single raw value into { label, value } format. Will be called by each value */
   getLabeledValue: GetLabeledValue<FlattenOptionsType<OptionsType>>;
   filterOptions: FilterOptions<OptionsType>;
-  findValueOption: // Need still support legacy ts api
+  findValueOption:  // Need still support legacy ts api
```
Comment (zombieJ) [inline src/generate.tsx:191]: No need 2 space comment

File: src/generate.tsx [hunk 2]
```diff
@@ -443,9 +445,12 @@ export default function generateSelector<
           key: '__RC_SELECT_TAG_PLACEHOLDER__',
         });
       }
+      if (filterSort && Array.isArray(filteredOptions)) {
+        filteredOptions.sort(filterSort);
```
Comment (zombieJ) [inline src/generate.tsx:449]: Create a new array instead for imutable

File: src/generate.tsx (hunk 1 above)
Comment (n0ruSh) [reply src/generate.tsx]: I have removed the spaces but it seems that they are added by `prettier` in `preCommit` hook.

**Review by zombieJ — APPROVED** [review]

**Comment by zombieJ** [comment]
Strange. CI failed. Let me check why...

**Comment by zombieJ** [comment]
@n0ruSh rebase master~

**Comment by zombieJ** [comment]
`+ rc-select@11.5.0`

Pls help to PR for antd feature branch : )

**Pull request merged**